src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java

```diff
@@ -457,10 +597,47 @@ public String getQuestionTypeChoiceOption() {
     @Override
     public List<String> validateQuestionDetails(String courseId) {
         List<String> errors = new ArrayList<>();
-        if (generateOptionsFor == FeedbackParticipantType.NONE
-                && numOfMcqChoices < Const.FeedbackQuestion.MCQ_MIN_NUM_OF_CHOICES) {
-            errors.add(Const.FeedbackQuestion.MCQ_ERROR_NOT_ENOUGH_CHOICES
-                       + Const.FeedbackQuestion.MCQ_MIN_NUM_OF_CHOICES + ".");
+        if (generateOptionsFor == FeedbackParticipantType.NONE) {
+
+            if (numOfMcqChoices < Const.FeedbackQuestion.MCQ_MIN_NUM_OF_CHOICES) {
+                errors.add(Const.FeedbackQuestion.MCQ_ERROR_NOT_ENOUGH_CHOICES
+                        + Const.FeedbackQuestion.MCQ_MIN_NUM_OF_CHOICES + ".");
+            }
+
+            // If weights are enabled, number of choices and weights should be same.
+            // In case if a user enters an invalid weight for a valid choice,
+            // the mcqChoices.size() will be greater than mcqWeights.size(), which will
+            // trigger this error message.
+            if (hasAssignedWeights && mcqChoices.size() != mcqWeights.size()) {
+                errors.add(Const.FeedbackQuestion.MCQ_ERROR_INVALID_WEIGHT);
+            }
+
+            // If weights are not enabled, but weight list is not empty or otherWeight is not 0
+            // In that case, this error will be triggered.
+            if (!hasAssignedWeights && (!mcqWeights.isEmpty() || mcqOtherWeight != 0)) {
+                errors.add(Const.FeedbackQuestion.MCQ_ERROR_INVALID_WEIGHT);
+            }
+
+            // If weight is enabled, but other option is disabled, and mcqOtherWeight is not 0
+            // In that case, this error will be triggered.
+            if (hasAssignedWeights && !otherEnabled && mcqOtherWeight != 0) {
+                errors.add(Const.FeedbackQuestion.MCQ_ERROR_INVALID_WEIGHT);
+            }
+
+            // If weights are enabled, and any of the weights have negative value, this error will
+            // be triggered.
```

Comment:
A note for future: usually this type of comments are written in imperative mood. 
e.g., `// If weights are enabled, and any of the weights have negative value, trigger this error`